frontend/src/metabase/query_builder/actions/core/parameterUtils.ts

```diff
@@ -29,33 +29,27 @@ type QueryParams = BlankQueryOptions & {
   objectId?: string;
 };
 
-function checkShouldPropagateDashboardParameters({
+function shouldPropagateDashboardParameters({
   cardId,
   deserializedCard,
   originalCard,
 }: {
   cardId?: number;
-  deserializedCard?: Card;
+  deserializedCard: Card;
   originalCard?: Card;
-}) {
-  if (!deserializedCard) {
-    return false;
-  }
-  if (cardId && deserializedCard.parameters) {
-    return true;
-  }
-  if (!originalCard) {
-    return false;
-  }
-  const equalCards = cardIsEquivalent(deserializedCard, originalCard, {
-    checkParameters: false,
-  });
-  const differentParameters = !cardIsEquivalent(
-    deserializedCard,
-    originalCard,
-    { checkParameters: true },
+}): boolean {
+  return Boolean(
+    // Either the card is saved and deserializedCard has parameters
+    (cardId && deserializedCard.parameters) ||
+      // Or deserializedCard is equivalent to originalCard, except for its parameters
+      (originalCard &&
+        cardIsEquivalent(deserializedCard, originalCard, {
+          checkParameters: false,
+        }) &&
+        !cardIsEquivalent(deserializedCard, originalCard, {
+          checkParameters: true,
+        })),
```

Comment:
This is exactly the same as before, except compressed into a boolean expression.

It's just missing the first check:
```
if (!deserializedCard) {
    return false;
  }
```

Reply:
This block I find more friendly in the original structure.

Particularly `return equalCards && differentParameters`, that I find highly declarative.

Reply:
+1 to Gus, we should avoid complex nested conditions like that one; I'd vote for breaking it down. Refactoring these actions for readability and untangling deep nesting came with a cost; we shouldn't get back to it again